pkg/3scale/amp/component/system.go

```diff
@@ -206,6 +218,18 @@ func (system *System) buildSystemBaseEnv() []v1.EnvVar {
 		helper.EnvVarFromSecret("MEMCACHE_SERVERS", SystemSecretSystemMemcachedSecretName, SystemSecretSystemMemcachedServersFieldName),
 	)
 
+	if system.Options.SystemDbTLSEnabled {
+		result = append(result,
+			helper.EnvVarFromSecretOptional("DB_SSL_CA", SystemSecretSystemDatabaseSecretName, SystemSecretSslCa),
+			helper.EnvVarFromSecretOptional("DB_SSL_CERT", SystemSecretSystemDatabaseSecretName, SystemSecretSslCert),
+			helper.EnvVarFromSecretOptional("DB_SSL_KEY", SystemSecretSystemDatabaseSecretName, SystemSecretSslKey),
+			helper.EnvVarFromSecretOptional("DATABASE_SSL_MODE", SystemSecretSystemDatabaseSecretName, SystemSecretDatabaseSslMode),
+			helper.EnvVarFromValue("DATABASE_SSL_CA", helper.TlsCertPresent("DATABASE_SSL_CA", SystemSecretSystemDatabaseSecretName, system.Options.SystemDbTLSEnabled)),
```

Comment:
as discussed, let's move the creation of empty keys + validation of keys + validation of values here: https://github.com/3scale/3scale-operator/blob/master/pkg/3scale/amp/operator/highavailability_options_provider.go#L29

Reply:
Agree its better if the reconciler catches this.

Reply:
Yes the secret validation should check if the secret values are populated in the highavailability reconcile and populate with an empty string if not set and return an error.
 
Two things here
- It doesn't look like highavailability handles zync database at the moment
- And may still leave the helper to populate the env vars that are not available in the secret .
i.e. the env vars in the secret have actual certs where as the env var that are used have paths to certs